Carry over release highlights from Lucene 9.0. (#83192)
elastic/elasticsearch: docs/reference/release-notes/highlights.asciidoc

// Use the notable-highlights tag to mark entries that 
// should be featured in the Stack Installation and Upgrade Guide:

// tag::notable-highlights[] 
[discrete]

> === Better space efficiency for `keyword`, `match_only_text`, and `text` fields

Reviewer: I think this better communicates the value (lower cost of ownership), but the current heading is fine.
```suggestion
=== Storage savings for `keyword`, `match_only_text`, and `text` fields
```

> The inverted index switched to a new encoding that is more space-efficient.

Reviewer: Some users may not be familiar with internal indices or know they're an internal data structure.
```suggestion
We've updated inverted indices, an internal data structure, to use a more space-efficient encoding.
```

> This change will benefit `keyword` fields, `match_only_text` fields and, to a

Reviewer: Grammar nit.
```suggestion
This change will benefit `keyword` fields, `match_only_text` fields, and, to a
```

> lesser extent, `text` fields. On the dataset of application logs that the {es}
> team uses for nightly benchmarks, this translated into a 14.4% reduction of the

Reviewer: I don't _think_ we need to mention nightly benchmarks, but ignore this if I'm wrong.
```suggestion
lesser extent, `text` fields. In our benchmarks using application logs, this translated into a 14.4% reduction of the
```

size of the index of the `message` field (mapped as `match_only_text`) and an
overall 3.5% reduction of the on-disk footprint.

This change will be picked up automatically by both new indices, and existing
indices for every new segment.

[discrete]

> === Faster indexing of `geo_point`, `geo_shape` and range fields

Reviewer: Minor grammar nit
```suggestion
=== Faster indexing of `geo_point`, `geo_shape`, and range fields
```

> Some optimizations have been made to indexing of multi-dimensional points,
> which are used by `geo_point` fields, `geo_shape` fields and range fields.

Reviewer: Rewords this to avoid passive voice. I also note that multi-dim points are an internal data structure.
```suggestion
We've optimized indexing speeds for multi-dimensional points, an
internal data structure used for `geo_point`, `geo_shape`, and range fields.
```

> Lucene-level benchmarks reported 10-15% faster indexing, so Elasticsearch
> indices and data streams that mostly consist of these field types may observe a
> non-negligible indexing speedup.

Reviewer: I reworded this to split a long sentence. Feel free to keep `Lucene-level` in if you feel that's needed or fair.
```suggestion
Our benchmarks reported 10-15% faster indexing for these fields types. {es}
indices and data streams that mostly consist of these fields may see noticeable
improvements to indexing speed.
```
